Release notes 66 (#1000)
pulumi/pulumi-hugo: themes/default/content/blog/pulumi-release-notes-66/index.md

---
title: "Jan. 2nd releases: Pulumi Packages support for plugins hosted anywhere and Pulumi Service 3rd party audit for secrets decryption"

# The date represents the post's publish date, and by default corresponds with
# the date this file was generated. Posts with future dates are visible in development,
# but excluded from production builds. Use the time and timezone-offset portions of
# of this value to schedule posts for publishing later.

> date: 2022-01-21T13:20:48-08:00

Reviewer: Don't forget to bump this to the current date when you're ready to merge (so it shows up at the top of the list).

allow_long_title: true

# Draft posts are visible in development, but excluded from production builds.
# Set this property to `false` before submitting your post for review.
draft: false

# Use the meta_desc property to provide a brief summary (one or two sentences)
# of the content of the post, which is useful for targeting search results or social-media
# previews. This field is required or the build will fail the linter test.
meta_desc: The latest Pulumi updates also include Pulumi import support for Kubernetes CRD, various improvements to Helm Release, and native ES Module support. 

> 

Reviewer: Right now, Hugo is building a `summary` (for display on the index page) dynamically using the `<!--more-->` separator, which is why you see the bullets there:

<img width="778" alt="image" src="https://user-images.githubusercontent.com/274700/150916693-474982fb-f263-4fb8-bcbb-54d76eb7d45a.png">

This is fine, but if you wanted to put something a little more concise and introductory in that space to show up before the Read More link, you could specify a `summary` in the frontmatter params:

```suggestion

summary: A few interesting lines that introduce the post. These will appear here on the blog home page.

```

Up to you, I just figured I'd mention it in case you wanted a little more control over that space.

# The meta_image appears in social-media previews and on the blog home page.
# A placeholder image representing the recommended format, dimensions and aspect
# ratio has been provided for you.

> meta_image: meta.png

Reviewer: Reminder we'll need an updated image before we merge.

# At least one author is required. The values in this list correspond with the `id`
# properties of the team member files at /data/team/team. Create a file for yourself
# if you don't already have one.
authors:
    - meagan-cojocar

# At least one tag is required. Lowercase, hyphen-delimited is recommended.
tags:
    - features

# See the blogging docs at https://github.com/pulumi/docs/blob/master/BLOGGING.md.
# for additional details, and please remove these comments before submitting for review.
---

> 
> ---

Reviewer: This is causing a horizontal rule I'm assuming was unintentional:

<img width="397" alt="image" src="https://user-images.githubusercontent.com/274700/150915199-3bbe7d73-460a-446e-b28f-37fc26b7189c.png">

```suggestion
```

> Over the holidays we have been releasing new features and improvements. Read on to learn about what's new this release!

Reviewer: Should there be an `in` in here? (If not, feel free to disregard!)

```suggestion
Over the holidays we have been releasing new features and improvements. Read on to learn about what's new in this release!
```

- Cloud Providers and Packages
  - [New `Command` package]({{<relref "/blog/pulumi-release-notes-66#new-command-package">}})
  - [Support pulumi import for Kubernetes CRDs]({{<relref "/blog/pulumi-release-notes-m66#support-pulumi-import-for-kubernetes-crds">}})
  - [Various improvements to Helm Release]({{<relref "/blog/pulumi-release-notes-m66#various-improvements-to-helm-release">}})
- Pulumi CLI and core technologies
  - [Support using native ES modules as Pulumi scripts]({{<relref "/blog/pulumi-release-notes-m66#support-using-native-es-modules-as-pulumi-scripts">}})
  - [Support packages with plugins hosted in any third-party location]({{<relref "/blog/pulumi-release-notes-m66#support-packages-with-plugins-hosted-in-any-third-party-location">}})
  - [State locking default enabled]({{<relref "/blog/pulumi-release-notes-m66#state-locking-default-enabled">}})
- Pulumi Service & Pulumi.com
  - [Third-party audit log tracking for secrets decryption]({{<relref "/blog/pulumi-release-notes-m66#third-party-audit-log-tracking-for-secrets-decryption">}})

<!--more-->

## Cloud Providers and Packages

### New `Command` package

> As part of creating or updating infrastructure, it is often necessary to run scripts and/or commands. In order to improve this experience we released a new [Pulumi Command package]({{<relref "/registry/packages/command">}}) in the Pulumi registry which enables users to run scripts locally or remotely on a target VM as part of the Pulumi resource lifecycle.

Reviewer: ```suggestion
As part of creating or updating infrastructure, it is often necessary to run scripts and/or commands. In order to improve this experience we released a new [Pulumi Command package]({{<relref "/registry/packages/command">}}) in the Pulumi Registry which enables users to run scripts locally or remotely on a target VM as part of the Pulumi resource lifecycle.
```

This new package is supported in all Pulumi languages.

The Command package supports quite a few common patterns involving local and remote scripts execution, such as:

> - [A simple local resource (random)]({{<relref "/registry/packages/command/#a-simple-local-resource-random">}})
> - [Remote provisioning of an EC2 instance]({{<relref "/registry/packages/command/#remote-provisioning-of-an-ec2-instance">}})
> - [Invoking a Lambda during a Pulumi deployment]({{<relref "/registry/packages/command/#invoking-a-lambda-during-pulumi-deployment">}})
> - [Using local.Command with curl to manage an external REST API]({{<relref "/registry/packages/command/#invoking-a-lambda-during-pulumi-deployment">}})
> - [Graceful cleanup of workloads in a Kubernetes cluster]({{<relref "/registry/packages/command/#graceful-cleanup-of-workloads-in-a-kubernetes-cluster">}})

Reviewer: ```suggestion
- [A simple local resource (random)]({{<relref "/registry/packages/command#a-simple-local-resource-random">}})
- [Remote provisioning of an EC2 instance]({{<relref "/registry/packages/command#remote-provisioning-of-an-ec2-instance">}})
- [Invoking a Lambda during a Pulumi deployment]({{<relref "/registry/packages/command#invoking-a-lambda-during-pulumi-deployment">}})
- [Using local.Command with curl to manage an external REST API]({{<relref "/registry/packages/command#invoking-a-lambda-during-pulumi-deployment">}})
- [Graceful cleanup of workloads in a Kubernetes cluster]({{<relref "/registry/packages/command#graceful-cleanup-of-workloads-in-a-kubernetes-cluster">}})
```

Reviewer 2: @cnunciato is this something we can lint for?

Reviewer: Not with anything we have now, unfortunately. There's a handful of other things we'll need to add along this same line, though (like checking for root-relative paths, etc.), so it's definitely on my radar. Probably something we'll need to write at least partly ourselves, though.

Reviewer: Filed this to track it: https://github.com/pulumi/pulumi-hugo/issues/1003

Reviewer 2: okay, would love to figure out if we can prioritize that kind of work in the future if possible to make our contribution and review process better.

Reviewer: > would love to figure out if we can prioritize that kind of work

Yep, it's in my queue, so it'll get looked at. :)

> [Learn more in this GitHub issue](https://github.com/pulumi/pulumi/issues/99)

Reviewer 2: ```suggestion
Learn more in the [Command package GitHub issue](https://github.com/pulumi/pulumi/issues/99)
```

Reviewer 2: I didn't add comments on all of these, but we should avoid including phrases like "learn more" in link text, and we should make the link text as specific as possible.

ref: https://webaim.org/techniques/hypertext/link_text

> [Learn more in this GitHub issue](https://github.com/pulumi/pulumi/issues/99)

Reviewer: Lil' nit, these should also end in periods (just 'cause end of sentence).

```suggestion
[Learn more in this GitHub issue](https://github.com/pulumi/pulumi/issues/99).
```

[Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes/pull/1818) and [this GitHub issue](https://github.com/pulumi/pulumi-kubernetes/pull/1828).

## Pulumi CLI and core technologies

### Support using native ES modules as Pulumi scripts

Native ECMAScript module (ESM) support has been added for the Node.js SDK. Pulumi users can now use Pulumi in projects with "type": "module" configured.

[Learn more in this GitHub issue](https://github.com/pulumi/pulumi/issues/7764)

### Support packages with plugins hosted in any third-party location

Pulumi Packages can now host their plugins anywhere (like GitHub releases) instead of needing to be published by Pulumi. We now detect any dependency that contains pulumi-plugin.json and treat it as a Pulumi Package, automatically downloading associated plugins as needed. To support this, the freshly generated Multi-Language Component (MLC) plugin will now include pulumi-plugin.json by default.

> Learn more in the following GitHub issues: [#8515](https://github.com/pulumi/pulumi/issues/8515), [#8516](https://github.com/pulumi/pulumi/issues/8516), [#8517](https://github.com/pulumi/pulumi/issues/8517), [#8530](https://github.com/pulumi/pulumi/issues/8530), [#8527](https://github.com/pulumi/pulumi/issues/8527) and [#8532](https://github.com/pulumi/pulumi/issues/8532).

Reviewer 2: these ones really need some kind of link text that isn't just the issue number, can we use the issue title instead?

### State locking default enabled

We [previously added](https://github.com/pulumi/pulumi/pull/2697) support for self-managed backend state locking behind the  PULUMI_SELF_MANAGED_STATE_LOCKING=1 flag. After positive feedback from users on this feature, we are making this the default when using a local or cloud backend such as Amazon S3, Google Cloud Storage and Azure Blob Storage. 

[Learn more in this GitHub issue](https://github.com/pulumi/pulumi/issues/8565)

> ## Pulumi Service & Pulumi.com

Reviewer: I'd remove this:

```suggestion
## Pulumi Service
```

... since there isn't anything having to do with the website ("pulumi.com") here.
